_posts/2018-11-12-cloud-cloud-computing.md

```diff
@@ -0,0 +1,89 @@
+
+The idea of the cloud is believed to have first been coined by [Joseph Carl Robnett Licklider](https://en.wikipedia.org/wiki/J._C._R._Licklider) in the 1960s.
+
+The cloud itself is indestructible because like we said earlier, it is a virtual space. The cloud is not physical.
+
+## Cloud computing.
+
+cloud computing is the delivery of cloud services.
+There are other types of computing such as [Distributed Computing](https://en.wikipedia.org/wiki/Distributed_computing),[Grid computing](https://en.wikipedia.org/wiki/Grid_computing),[Utility Computing](http://en.wikipedia.org/wiki/Utility_computing), and [Cluster Computing](http://en.wikipedia.org/wiki/Computer_cluster).
+
+Cloud services are delivered in a number of ways; Infrastructure as a service(IaaS), Platform as a service(PaaS) and Software as a service(SaaS). Here is a picture of resources you would traditionally manage.
+
+![Traditional IT model](/assets/images/blog/cloud-computing/traditional-IT.png){:class="img-responsive center"}
+
+### IaaS
+
+IaaS stands for Infrastructure as a service. With this service model, what you get, in simple terms, is a data center. Some aspects of computing are managed for  you by your service provider.
+
+Here's a picture.
+![IaaS model](/assets/images/blog/cloud-computing/Iaas.png){:class="img-responsive center"}
+
+
+
+
+From the image, you can see that  networking, storage, servers, virtualization and part of the Operating system are provided as a service.
+Examples Iaas providers include; [Digital Ocean](https://www.digitalocean.com/),[linode](https://www.linode.com/), [AWS](https://aws.amazon.com/),[vultr](https://www.vultr.com/).
+
+Say you want a Linux based server. You will go to any of the above sites. Provision yourself a virtual machine and that's it. You can now deploy whatever application you want to the machine/server, in the cloud. You can make your application accessible to anyone over the internet.
+Virtualization software, storage for your application and related libraries, operating system and networking are all provided, in the cloud. 
+
+
+You will, however, have to manage your own operating system, provide middleware example an antivirus, provide your application with data and management system for it, if your application needs a database you would have to create it, provide runtime libraries and the actual application.
+
+
+
+### PaaS
+
+Paas stands for Platform as a Service. The service provider provides you with a pre-configured infrastructure.PaaS includes middleware, runtimes, development, testing and deployment tools.
+Here's a picture;
+
+![PaaS model](/assets/images/blog/cloud-computing/paas.png){:class="img-responsive center"}
+
+
+
+All you provide is code for your application and the data.
+Example of Paas providers are [Heroku](https://www.heroku.com/), [Go daddy](https://uk.godaddy.com), etc
```

Comment:
Heroku is clear but godaddy is confusing. Please link to the exact product that is a PaaS. Also mention BaaS as well as opensource solution for running PaaS. FIrebase is also an example. List others here as well.

Reply:
Soem open source solution for PaaS are like flynn, dokku, deis etc. Please uncover all options.

Reply:
Also please list top PaaS that are used to deploy rails application specifically.

Reply:
Please include the open source options of PaaS as i had mentioned like flynn, dokku, deis